motioneye/mjpgclient.py

```diff
@@ -194,81 +207,78 @@ def _seek_http(self) -> None:
         future.add_done_callback(self._on_http)
 
     def _on_http(self, future: Future) -> None:
-        try:
-            data = future.result()
-        except Exception as e:
-            self._error(e)
-        else:
-            if data.endswith(b'401 '):
-                self._seek_www_authenticate()
+        result, data = self._get_future_result(future)
+        if not result:
+            return
 
-            else:  # no authorization required, skip to content length
-                self._seek_content_length()
+        if data.endswith(b'401 '):
+            self._seek_www_authenticate()
+
+        else:  # no authorization required, skip to content length
+            self._seek_content_length()
 
     def _seek_www_authenticate(self) -> None:
         future = utils.cast_future(self.read_until(b'WWW-Authenticate:'))
         future.add_done_callback(self._on_before_www_authenticate)
 
     def _on_before_www_authenticate(self, future: Future) -> None:
-        try:
-            future.result()
-        except Exception as e:
-            self._error(e)
-        else:
-            if self._check_error():
-                return
+        result, _ = self._get_future_result(future)
+        if not result:
+            return
 
-            r_future = utils.cast_future(self.read_until(b'\r\n'))
-            r_future.add_done_callback(self._on_www_authenticate)
+        if self._check_error():
+            return
+
+        r_future = utils.cast_future(self.read_until(b'\r\n'))
+        r_future.add_done_callback(self._on_www_authenticate)
 
     def _on_www_authenticate(self, future: Future) -> None:
-        try:
-            data = future.result()
-        except Exception as e:
-            self._error(e)
-        else:
-            if self._check_error():
-                return
+        result, data = self._get_future_result(future)
+        if not result:
+            return
+
+        if self._check_error():
+            return
```

Comment:
```suggestion
        if not result or self._check_error():
            return
```
If hope it is evaluated as
```py
if (not result) or self._check_error()
```
instead of
```py
if not (result or self._check_error())
```
😅

Reply:
Your hope is right :) hrmh... Damn I let lot's of these go through...